src/Common/AbstractEnum.php

```diff
@@ -0,0 +1,359 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Common;
+
+use BadMethodCallException;
+use InvalidArgumentException;
+use ReflectionClass;
+
+/**
+ * Abstract base class for enum-like behavior in PHP 7.4
+ *
+ * This class provides enum-like functionality for PHP versions that don't support native enums.
+ * Child classes should define uppercase snake_case constants for enum values.
+ *
+ * @example
+ * class PersonEnum extends AbstractEnum {
+ *     public const FIRST_NAME = 'first';
+ *     public const LAST_NAME = 'last';
+ * }
+ *
+ * // Usage:
+ * $enum = PersonEnum::from('first'); // Creates instance with value 'first'
+ * $enum = PersonEnum::tryFrom('invalid'); // Returns null
+ * $enum = PersonEnum::firstName(); // Creates instance with value 'first'
+ * $enum->name; // 'FIRST_NAME'
+ * $enum->value; // 'first'
+ * $enum->equals('first'); // Returns true
+ * $enum->is(PersonEnum::firstName()); // Returns true
+ * PersonEnum::cases(); // Returns array of all enum instances
+ *
+ * @since n.e.x.t
+ */
+abstract class AbstractEnum
+{
+    /**
+     * @var string|int The value of the enum instance
+     */
+    private $value;
+
+    /**
+     * @var string The name of the enum constant
+     */
+    private string $name;
+
+    /**
+     * @var array<string, array<string, string|int>> Cache for reflection data
+     */
+    private static array $cache = [];
+
+    /**
+     * @var array<string, array<string, self>> Cache for enum instances
+     */
+    private static array $instances = [];
+
+    /**
+     * Constructor is private to ensure instances are created through static methods
+     *
+     * @since n.e.x.t
+     * @param string|int $value The enum value
+     * @param string $name The constant name
+     */
+    final private function __construct($value, string $name)
+    {
+        $this->value = $value;
+        $this->name = $name;
+    }
+
+    /**
+     * Magic getter to provide read-only access to properties
+     *
+     * @since n.e.x.t
+     * @param string $property The property name
+     * @return mixed
+     * @throws BadMethodCallException If property doesn't exist
```

Comment:
In WordPress, the `@since` annotation is always part of another "group" of annotations, I.e. separated with a blank line. Is that something that conflicts with PSR or not? If not, might I suggest we do that here too? I do personally think that makes it easier to read, and it makes sense to separate, since `@param`, `@return`, and `@throws` are arguably the most crucial things to understand about a method.
```suggestion
     * @since n.e.x.t
     *
     * @param string $property The property name
     * @return mixed
     * @throws BadMethodCallException If property doesn't exist
```

If you agree, let's do that throughout.

Reply:
Sure! Resolved in [5cc061f](https://github.com/WordPress/php-ai-client/pull/25/commits/5cc061f0b33b9a9ab8f65db753547a2823ccb43d)